src/core/utils/Vector.hpp

```diff
@@ -287,6 +287,12 @@ template <size_t N, typename T> Vector<T, N> sqrt(Vector<T, N> const &a) {
   return ret;
 }
 
+template <class T>
+Vector<T, 3> vector_product(const Vector<T, 3> &a, const Vector<T, 3> &b) {
```

Comment:
east const :fist:

Reply:
![polyconst braclets](https://pbs.twimg.com/media/DfaTeeKXcAEGaig?format=jpg&name=small)

Reply:
We should decide on a convention though. I think `west const` is more what people are used to?

Reply:
there are many opinions on that: I'm in favor of east since ```const``` modifies the type on it's left unless there is none

Reply:
With west const the rule is `const modifies what is on its left`

Reply:
With `east const` the rule is `const modifies what is on its left`, if there is nothing, it's not `east const`. I like this argument for `east const`: [A foolish consistency](http://slashslash.info/2018/02/a-foolish-consistency/). The core guidline [say](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl26-use-conventional-const-notation) `west const`, but with weak arguments.